plan/milestones.md

```diff
@@ -0,0 +1,88 @@
+- An actual FL job. Set the FL hyperparameters and execute the actual job. 
+
+We describe the key milestones in detail below. 
+
+
+## Machine learning model 
+
+The prerequisite of a _federated_ learning job is a _machine_ learning model, which can be trained in a non-federated scheme. Such a model can be  
+- a working model trained with some real yet small data, or  
+- a model architecture that is proven effective via synthetic training data. 
+
+One example of a hospital (i.e., the customer) is described below. The hospital’s task is to classify a medical image into positive or negative of a specific disease. The data scientists at the hospital already have a neural network model trained in a centralized fashion with their own patient data. The model achieved reasonable performance. Now the hospital wants to further improve the model's performance as more hospitals would like to participate in a federated learning way without their data leaving their cluster. 
+
+## Resource provision 
+
+If all the data live in __one__ AAD (Azure Active Directory) tenant, we could simply create “vanilla” AML computes and easily use managed identity to enforce silos (via compute --> storage access). Briefly speaking, per-silo storage accounts should be locked down with only RBAC for access and don’t give anyone access keys, and per-silo compute should have a managed identity, give just that managed identity the "blob storage contributor" roles to access the data. For further details, please check out this [public document](https://docs.microsoft.com/en-us/azure/machine-learning/how-to-use-managed-identities?tabs=python) on managed identity.
+
+If __multiple__ AAD tenants are involved, we provide the following resource provision approach. Specifically, arc-enabled k8s compute clusters are used, since managed identity won’t work for cross-tenants scenarios. Please refer to [this file](https://github.com/Azure-Samples/azure-ml-federated-learning/tree/main/automated_provisioning) for more information. 
+
+1. There should be computes in each of the silos. These computes should all be in their own K8s cluster. If not, we provide [this script](https://github.com/Azure-Samples/azure-ml-federated-learning/blob/main/automated_provisioning/ps/CreateK8sCluster.ps1) to create a K8s cluster. 
+2. Provision an orchestrator workspace, and attach all the silo computes/k8s clusters via Azure Arc. We provide [this script](https://github.com/Azure-Samples/azure-ml-federated-learning/blob/main/automated_provisioning/ps/ConnectSiloToOrchestrator.ps1) to connect the silos to the orchestrator here.  
+3. We also provide a mock [script](https://github.com/Azure-Samples/azure-ml-federated-learning/blob/main/automated_provisioning/sample_job/RunSampleJob.ps1) here that does the fan-out/fan-in and uses the silos to ensure the orchestrator can access the k8s in each silo. 
```

Comment:
@jialei777  I think the "RunSampleJob.ps1" script here only run a simple job on the arc-enabled k8s compute cluster of a silo, and it is not a full fl pipeline that does fan-out/fan-in operations between orchestrator and silos. So the wordings "that does the fan-out/fan-in" might be inaccurate.

Suggest modifying the sentence here a bit to sth like "We also provide a mock [script](https://github.com/Azure-Samples/azure-ml-federated-learning/blob/main/automated_provisioning/sample_job/RunSampleJob.ps1) here to help validate that the connected arc-enabled k8s compute cluster in each silo is working properly."